src/wallet/wallet.cpp

```diff
@@ -1927,26 +1928,35 @@ bool CWallet::FundTransaction(CMutableTransaction& tx, CAmount &nFeeRet, int& nC
 
     CReserveKey reservekey(this);
     CWalletTx wtx;
-    if (!CreateTransaction(vecSend, wtx, reservekey, nFeeRet, nChangePosRet, strFailReason, &coinControl, false))
+    if (!CreateTransaction(vecSend, wtx, reservekey, nFeeRet, nChangePosInOut, strFailReason, &coinControl, false))
         return false;
 
-    if (nChangePosRet != -1)
-        tx.vout.insert(tx.vout.begin() + nChangePosRet, wtx.vout[nChangePosRet]);
+    if (nChangePosInOut != -1)
+        tx.vout.insert(tx.vout.begin() + nChangePosInOut, wtx.vout[nChangePosInOut]);
```

Comment:
@promag: The check should should really be here (maybe an additional check). If someone uses `CWallet::FundTransaction()` with a `nChangePosInOut >= wtx.vout.size()`, the app crashes/throws an exceptions. Future changes might easy get trapped by this.

The problematic part is `wtx.vout[nChangePosInOut]`. There should be a check somewhere in `CWallet::FundTransaction()` that leads to a `return false` if a overflow has detected.